pmd-java/src/main/java/net/sourceforge/pmd/lang/java/metrics/impl/visitors/CycloVisitor.java

```diff
@@ -0,0 +1,160 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.lang.java.metrics.impl.visitors;
+
+import org.apache.commons.lang3.mutable.MutableInt;
+
+import net.sourceforge.pmd.lang.java.ast.ASTAssertStatement;
+import net.sourceforge.pmd.lang.java.ast.ASTBlockStatement;
+import net.sourceforge.pmd.lang.java.ast.ASTCatchStatement;
+import net.sourceforge.pmd.lang.java.ast.ASTConditionalExpression;
+import net.sourceforge.pmd.lang.java.ast.ASTDoStatement;
+import net.sourceforge.pmd.lang.java.ast.ASTForStatement;
+import net.sourceforge.pmd.lang.java.ast.ASTIfStatement;
+import net.sourceforge.pmd.lang.java.ast.ASTSwitchLabel;
+import net.sourceforge.pmd.lang.java.ast.ASTSwitchStatement;
+import net.sourceforge.pmd.lang.java.ast.ASTThrowStatement;
+import net.sourceforge.pmd.lang.java.ast.ASTWhileStatement;
+import net.sourceforge.pmd.lang.java.ast.JavaNode;
+import net.sourceforge.pmd.lang.java.ast.JavaParserVisitorAdapter;
+import net.sourceforge.pmd.lang.java.metrics.impl.CycloMetric;
+import net.sourceforge.pmd.lang.java.metrics.impl.CycloMetric.CycloOption;
+import net.sourceforge.pmd.lang.metrics.MetricOptions;
+
+
+/**
+ * Visitor for the Cyclo metric.
+ *
+ * @author Clément Fournier
+ * @since 6.7.0
+ */
+public class CycloVisitor extends JavaParserVisitorAdapter {
```

Comment:
I guess, this class itself is internal (it's inside a subpackage of a package called `impl`).
Should we annotate this then with `@InternalApi` as well?